tests/e2e/nvidia-container-toolkit_test.go

```diff
@@ -24,7 +24,27 @@ import (
 )
 
 // Integration tests for Docker runtime
-var _ = Describe("docker", func() {
+var _ = Describe("docker", Ordered, func() {
+	var s Script
+
+	// Install the NVIDIA Container Toolkit
+	BeforeAll(func(ctx context.Context) {
+		switch {
+		case sshKey == "":
+			s = localRun{}
+		default:
+			s = remoteRun{sshKey, sshUser, remoteHost}
+		}
+
+		if installCTK {
+			installer, err := newInstaller(dockerInstallTemplate, imageRepo+":"+imageTag, sshKey, sshUser, remoteHost)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = installer.Install()
+			Expect(err).ToNot(HaveOccurred())
+		}
```

Comment:
The logic on which `Script` (or `Runner`) should be used should be pulled into a constructor with functional arguments. This instance should also be reused in the `Installer` so as to not duplicate the logic there.
```suggestion
		s := NewRunner(
			WithHost(remoteHost),
			WithSshKey(sshKey),
			WithSshUser(sshUser),
		)

		if installCTK {
			installer, err := NewToolkitInstaller(
				WithRunner(s),
				WithVersion(version),
			)
			Expect(err).ToNot(HaveOccurred())

			err = installer.Install()
			Expect(err).ToNot(HaveOccurred())
		}
```